go/arrow/csv/reader.go

```diff
@@ -469,6 +469,10 @@ func (r *Reader) initFieldConverter(bldr array.Builder) func(string) {
 		return func(s string) {
 			r.parseList(bldr, s)
 		}
+	case arrow.BinaryDataType:
+		return func(s string) {
+			r.parseBinaryDataType(bldr, s)
```

Comment:
Like the string case, we probably want to check the `r.stringsCanBeNull` value to determine whether we want to check for nulls or just let it be empty values etc.

Reply:
I handled it a bit differently as `""` shouldn't mean null in base64 but rather just empty and `(null)` is not a valid base64 encoding anyway so we don't really need to check the `r.StringsCanBeNull`